app/models/aliquot.rb

```diff
@@ -59,4 +59,43 @@ def tagged?
   def collection?
     false
   end
+
+  # Returns the id to publish
+  def publish_id
+    id.to_s
+  end
+
+  # Returns the type of the source to publish
+  def publish_source_type
+    source.respond_to?(:publish_data_source) ? source.publish_data_source[:sourceType].to_s : ''
+  end
+
+  # Returns the barcode of the source to publish
+  def publish_source_barcode
+    source.respond_to?(:publish_data_source) ? source.publish_data_source[:sourceBarcode].to_s : ''
+  end
+
+  # Returns the sample name of the source to publish
+  def publish_sample_name
+    if source.respond_to?(:publish_data_source)
+      source.publish_data_source[:sampleName].to_s
+    else
+      ''
+    end
+  end
+
+  # Returns the barcode of the used_by to publish
```

Comment:
Same again. Not sure why we need the respond to and it feels like anything to do with publishing should be in an object or nested class.